### PR TITLE
Updates for Chef 17

### PR DESCRIPTION
**Pull request by ramereth**
- Add support for Chef 17 by enabling `unified_mode` to resources
- Increase required Chef Infra Client version to >= 15.3
- Add `kitchen.dokken.yml` and enable Github CI testing
- Convert ServerSpec to InSpec

Signed-off-by: Lance Albertson <lance@osuosl.org>

### CODE REVIEW

**Comment by ramereth** [comment]
This supersedes #95

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+      - master
+
+jobs:
+  delivery:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out code
+        uses: actions/checkout@master
+      - name: Run Chef Delivery
+        uses: actionshub/chef-delivery@master
```
Comment (xorima) [inline .github/workflows/ci.yml]:
```suggestion
        uses: actionshub/chef-delivery@main
```

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        suite:
+          - 'default'
+          - 'secret-resource'
+      fail-fast: false
+
+    steps:
+      - name: Check out code
+        uses: actions/checkout@master
+      - name: Install Chef
+        uses: actionshub/chef-install@master
```
Comment (xorima) [inline .github/workflows/ci.yml]:
```suggestion
        uses: actionshub/chef-install@main
```

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+      - name: Dokken
+        uses: actionshub/kitchen-dokken@master
```
Comment (xorima) [inline .github/workflows/ci.yml]:
```suggestion
        uses: actionshub/kitchen-dokken@main
```

**Review by xorima — CHANGES_REQUESTED** [review]

**Pull request closed without merging**